src/lib/expansion-panel/expansion-panel.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  Component,
+  Directive,
+  EventEmitter,
+  Input,
+  Output,
+  ViewEncapsulation,
+  AfterContentInit,
+  OnDestroy,
+  ContentChildren,
+  QueryList,
+  Optional,
+  forwardRef,
+} from '@angular/core';
+import {
+  trigger,
+  state,
+  style,
+  transition,
+  animate,
+} from '@angular/animations';
+import {Subscription} from 'rxjs/Subscription';
+
+/** Used to generate unique ID for each expansion panel. */
+let nextId = 0;
+/** Time and timing curve for expansion panel animations. */
+export const EXPANSION_PANEL_ANIMATION_TIMING = '225ms cubic-bezier(0.4,0.0,0.2,1)';
+
+@Directive({
+  selector: 'md-expansion-panel-set, mat-expansion-panel-set',
+  host: {
+    '[class.mat-expansion-panel-set]': 'true',
+  },
+})
+export class MdExpansionPanelSet implements AfterContentInit, OnDestroy {
+  /** Whether the panel set should use flat styling. */
+  @Input('style') style: 'default' | 'flat' | 'wide' = 'default';
+  /** Whether the panel set should allow multiple open panels. */
+  @Input('multi') multi: boolean = false;
+  /** A list of subscriptions for panel open events. */
+  private _panelSubscriptions: Subscription[] = [];
+  /** QueryList of all expansion panels in the expansion panel set. */
+  @ContentChildren(forwardRef(() => MdExpansionPanel)) private _panels: QueryList<MdExpansionPanel>;
+
+  /** Set up event subscriptions for all panels for when they open, closing the other panels. */
+  ngAfterContentInit() {
+    this._panels.forEach((panel: MdExpansionPanel, idx: number, panels: MdExpansionPanel[]) => {
+      let panelId = panel.getExpansionPanelId();
+      let otherPanels = panels.filter(otherPanel => otherPanel.getExpansionPanelId() !== panelId);
+      let panelSubscription = panel.open.asObservable().subscribe(() => {
+        if (!this.multi) {
+          otherPanels.forEach(otherPanel => otherPanel.expanded = false);
+        }
+      });
+      this._panelSubscriptions.push(panelSubscription);
+    });
+  }
+
+  /** Clean up panel subscriptions. */
+  ngOnDestroy() {
+    this._panelSubscriptions.forEach(subscription => subscription.unsubscribe());
```

Comment:
Instead of saving multiple subscriptions in an array and dealing with them individually, you can use `Observable.merge` to merge them into a single stream.